### PR TITLE
ENH: add MAGICs

### PR DESCRIPTION
**Pull request by tacaswell**
Add a helper function that registers an IPython magics to run call
the run engine.

This function can/should be called in the beamline configuration
once the 'canonical' runengine has been set up.

This is support work for a coming PR from @danielballan

### CODE REVIEW

File: bluesky/utils.py [hunk 1]
```diff
@@ -585,3 +585,38 @@ def _qt_kicker():
...
+            # You must call the parent constructor
+            super().__init__(shell)
+            self.RE = RE
+
+        @line_magic
+        def RE(self, line):
+            "my line magic"
+            if line:
+                inp = eval(line, self.shell.user_ns)
+                self.RE(inp)
```
Comment (klauer) [inline bluesky/utils.py]: `ret = `?

Comment (tacaswell) [inline bluesky/utils.py]: yup

**Comment by klauer** [comment]
My alternative suggestion:

```python
import IPython
from IPython.core.inputtransformer import StatelessInputTransformer


@StatelessInputTransformer.wrap
def tr_run_engine(line):
    if line.startswith('>>'):
        line = line[2:].strip()
        return 'RE({})'.format(line)
    return line


def RE(value):
    print('runengine!', value)


ip = IPython.get_ipython()
ip.input_splitter.logical_line_transforms.append(tr_run_engine())
ip.input_transformer_manager.logical_line_transforms.append(tr_run_engine())
```

Result:

```python
In [4]: >> 'test'
runengine! test

In [5]: foo = 'a'

In [6]: >> foo
runengine! a
```

**Comment by klauer** [comment]
This default input transformer is also rather interesting:
https://github.com/ipython/ipython/blob/master/IPython/core/inputtransformer.py#L39

```python
In [1]: def fcn(*args, **kwargs):
   ....:     print('args is', args, 'kwargs is', kwargs)

In [2]: /fcn 1 2 3 a=4 b=5 c=6
args is (1, 2, 3) kwargs is {'b': 5, 'c': 6, 'a': 4}
```

**Comment by danielballan** [comment]
I am skeptical that this is actually better than convincing people to type `RE(...)` in the end, but I think it is a worthwhile experiment. I am in favor of merging it and seeing how people like it.

**Comment by danielballan** [comment]
Why not just a single `<`? There is still no ambiguity:

```python
In [1]: < 1
  File "<ipython-input-1-732f6b4d3ec1>", line 1
    < 1
    ^
SyntaxError: invalid syntax
```

**Comment by ericdill** [comment]
>  am skeptical that this is actually better than convincing people to type RE(...) in the end

I totally agree.  @yugangzhang seemed pretty happy to just throw plans at the run engine via the RE() syntax, which feels like big praise to the notion of plans.

> but I think it is a worthwhile experiment. I am in favor of merging it and seeing how people like it

I'm pretty concerned that this experiment is going to result in bug reports like:

```python
def foo():
    < ct()
    < ct()
    < ct()
```

and essentially cause more work in the short term.  That being said, experiments are always a good thing when we are trying to converge on the right user interface here and I think that doing this experiment will ultimately result in a better user interface.

Interestingly, I initially started this comment strongly in the "don't do this. this will result in more work" camp, but through the process of typing out my reasoning, I have convinced myself that it is worth doing because even though I think it will ultimately fail, we will learn important things about what the users actually want and is thus very worth doing.

**Comment by danielballan** [comment]
Thanks for the comments, @ericdill. Now that we've all slept on this for awhile and the related SPEC API change is about ready to go, it's time to make a decision. Where is everyone standing on this?

**Comment by danielballan** [comment]
Signs already point to this being maybe a bad idea, but I think we all agree that it might be worth trying. I'm merging it so we have it handy if/when this comes up.

**Comment by klauer** [comment]
I'm just glad `git blame` points to @tacaswell. 👍

**Pull request merged**